packages/client/ui/react-ui/src/providers/CrossmintWalletProvider.tsx

```diff
@@ -44,10 +46,25 @@ type WalletContext = {
     getOrCreateWallet: (
         config?: Pick<WalletConfig, "signer" | "type">
     ) => Promise<{ startedCreation: boolean; reason?: string }>;
-    createPasskeySigner: () => Promise<PasskeySigner | null>;
+    createPasskeySigner: (name: string) => Promise<PasskeySigner | null>;
     clearWallet: () => void;
 };
 
+type WalletCache = {
```

Comment:
We should extract all new cache and passkey functionality as this provider is already getting too big and that way we can make the provider itself more declarative and easier to read

Reply:
@AlbertoElias agree the file became too large.

Moved all the persistance stuff into a new `useWalletCache` hook.